packages/dma-contracts/contracts/v2/OperationExecutor.sol

```diff
@@ -0,0 +1,195 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.15;
+
+import { OperationsRegistryV2 } from "./OperationsRegistry.sol";
+import { ServiceRegistry } from "../core/ServiceRegistry.sol";
+import { StorageSlot } from "./UseStorageSlot.sol";
+import { ActionAddress } from "../libs/ActionAddress.sol";
+import { TakeFlashloan } from "../actions/common/TakeFlashloan.sol";
+import { Executable } from "../actions/common/Executable.sol";
+import { IERC3156FlashBorrower } from "../interfaces/flashloan/IERC3156FlashBorrower.sol";
+import { IERC3156FlashLender } from "../interfaces/flashloan/IERC3156FlashLender.sol";
+import { IFlashLoanRecipient } from "../interfaces/flashloan/balancer/IFlashLoanRecipient.sol";
+import { SafeERC20, IERC20 } from "../libs/SafeERC20.sol";
+import { SafeMath } from "../libs/SafeMath.sol";
+import { FlashloanData, Call } from "../core/types/Common.sol";
+
+error UntrustedLender(address lender);
+error UnknownActionsSet(bytes packedActionHashes);
+error InconsistentAsset(address flashloaned, address required);
+error InconsistentAmount(uint256 flashloaned, uint256 required);
+error ForbiddenCall(address caller);
+
+interface IProxy {
+  function execute(address, bytes memory) external payable returns (bytes memory);
+}
+
+interface IOperationStorage {
+  function setInitiator(address) external;
+}
+
+/**
+ * @title Operation Executor
+ * @notice Is responsible for executing sequences of Actions (Operations)
+ */
+
+contract OperationExecutorV2 is IERC3156FlashBorrower, IFlashLoanRecipient {
+  using ActionAddress for address;
+  using SafeERC20 for IERC20;
+  using SafeMath for uint256;
+
+  ServiceRegistry immutable REGISTRY;
+  OperationsRegistryV2 immutable OPERATIONS_REGISTRY;
+  address immutable FLASH_MINT_MODULE;
+  address immutable BALANCER_VAULT;
+  address immutable OPERATION_EXECUTOR;
+
+  /**
+   * @dev Emitted once an Operation has completed execution
+   * @param name The address initiating the deposit
```

Comment:
dont use chatGPT to create docs :D

Reply:
This is something that's been there. I haven't read the description :Rofl: